onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -742,9 +742,20 @@ class PlannerImpl {
             }
           }
         } else if (IsNonTensor(*node_output)) {
-          // we do not try sharing-optimization for non-tensors
-          AllocPlan(current).alloc_kind = AllocKind::kAllocate;
-          AllocPlan(current).program_counter.AddStart(program_counter);
+          // For optional type outputs, certain kernels allow re-using input OrtValues.
+          // Check if re-using is feasible.
+          if (!context_.IsParallelExecutionEnabled() &&
+              IsOptionalType(*node_output) &&
```

Comment:
Can we pull this up the level in if/else so it is a separate clause, instead of putting everything under NonTensor(). You actually decalared it as a separate OptionalBase in types, not NonTensor

Reply:
done